content/php/concepts/arrays/terms/array-push/array-push.md

```diff
@@ -0,0 +1,111 @@
+---
+Title: 'array_push()'
+Description: 'Inserts one or more element values to the end of an array, and returns the new number of element values in the array.'
+Subjects:
+  - 'Computer Science'
+  - 'Web Design'
+  - 'Web Development'
+Tags:
+  - 'Arrays'
+  - 'Functions'
+CatalogContent:
+  - 'learn-php'
+  - 'paths/computer-science'
+---
+
+The **`array_push()`** method inserts one or more element values to the end of an array, and returns the new number of element values in the array.
+
+## Syntax
+
+```pseudo
+array_push($array, $value1, $value2, ... $valueN)
+```
+
+The `array_push()` function has one required parameter and some optional parameters:
+
+- `$array`: Specifies the input `array`.
+- `$value1 ... $valueN`: Specifies the element values to add. This function adds one or more element values at a time which are separated by commas.
+
+The `array_push()` function returns the element count of `$array` with the new element values inserted to the end of the `$array`.
+
+## Example
+
+The following example uses the `array_push()` function to add `"Chervolet"`, and `"Toyota"` element values to the end of the indexed array `$car_brands`:
+
+```php
+<?php
+$car_brands = array("Mercedes","Ford");
+array_push($car_brands,"Chervolet","Toyota");
+print_r($car_brands);
+?>
+```
+
+The example will result in the following output:
+
+```shell
+Array (
+[0] => Mercedes
+[1] => Ford
+[2] => Chervolet
+[3] => Toyota
+)
+```
+
+## Codebyte Example
+
+This example is runnable and uses the `array_push()` function:
+
+> **Note:** Each example below uses `print_r`, `echo`, and `foreach` syntaxes to output a specific array's values and keys after that array has been inserted with new element values using the `array_push()` function.  
+
+```codebyte/php
+<?php
```

Comment:
@phuongtrieu97coder Is there some problem you have with using the return value of the function in the example? In any event, since you didn't go with my original suggestion, can you prune this codebyte back to just the $supermarkets example?

Reply:
@SSwiniarski  Is there any problem with my syntaxes that you can't run it effectively? If not, please keep it. I just want users or students learn different ways to use array_push() function, I think you want it too. If you want the codebyte followed your rules, at least give me some examples about how you want the return values look like including the array_push() function, values inside it, the output syntax, the result you got, etc.

Reply:
Do something like this:
````
```codebyte/php
<?php
  $supermarkets = array("Costco","Walmart");
  $size = count($supermarkets);
  echo("Array legnth before array_push(): " . $size) . "\n\n";
  $newsize = array_push($supermarkets,"Sam's Club","Target");
  echo("Array legnth after array_push(): " . $newsize . "\n\n");
?>
```
````

Reply:
@SSwiniarski Could I add foreach loop into your examples?

Reply:
@phuongtrieu97coder Go ahead it's your entry. I just wanted the codebyte to be different from the original example and thought using the return value from the function was a good way to do that.